bach/bach/series/series.py

```diff
@@ -858,6 +877,28 @@ def sort_values(self, *, ascending=True):
             return self
         return self.copy_override(sorted_ascending=ascending)
 
+    def sort_by_series(self, by: List['Series'], ascending: Union[bool, List[bool]] = True):
```

Comment:
```suggestion
    def sort_by_series(self: T, by: List['Series'], ascending: Union[bool, List[bool]] = True) -> T:
```
Just a bit nicer for calling code

Reply:
I think we should do this stuff in a separate PR, there are a lot of functions in Series class that should be correctly typed. TBH I just followed the current pattern

Reply:
https://github.com/objectiv/objectiv-analytics/issues/1032

Reply:
another thing: maybe for consistency we should put an `*, ` before `ascending`. We have that for sort_index and sort_values

Reply:
> another thing: maybe for consistency we should put an `*, ` before `ascending`. We have that for sort_index and sort_values

Then we should follow this for `DataFrame` as well then. IMHO we should create a guide with this type of practices when contributing to Bach, as this type of nitpicking can be avoided in a future

Reply:
typing for the method here https://github.com/objectiv/objectiv-analytics/pull/995/commits/b76152d17cad2ab0cebe2cfcd3ef2663eec66001

Reply:
PR for rest of typings: https://github.com/objectiv/objectiv-analytics/pull/1038

Reply:
> Then we should follow this for DataFrame as well then. IMHO we should create a guide with this type of practices when contributing to Bach, as this type of nitpicking can be avoided for new contributors.

Totally okay with me not to do this change. It's fully arbitrary. I was looking at the other function definitions, and I saw we did it like that there, and figured I point it out. Didn't mean to imply this is a change that would be a blocker for me or any of the sort